pkg/cache/controller_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"testing"
+
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestClusterIndexFunc(t *testing.T) {
+	tests := []struct {
+		obj     *corev1.ConfigMap
+		desired string
+	}{
+		{obj: &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{ClusterName: "test"}}, desired: "test//"},
```

Comment:
@sttts do we want to add helpers to kcp-dev/logicalcluster to set the cluster name on an object? Thinking about preparing for ObjectMeta.ClusterName going away and not wanting to have to make multiple changes if we ever move cluster name again.

Reply:
I think this would make a lot of sense